src/storage/src/storage_state.rs

```diff
@@ -224,7 +224,10 @@ impl<'w, A: Allocate> Worker<'w, A> {
             object_status_updates: Default::default(),
             internal_cmd_tx: command_sequencer,
             async_worker,
-            storage_configuration: StorageConfiguration::new(connection_context),
+            storage_configuration: StorageConfiguration::new(
+                connection_context,
+                mz_dyncfgs::all_dyncfgs(),
```

Comment:
For dep reasons, ideally the only thing that depends on `mz_dyncfgs` is the very leaf `fn main()`s. They each already construct one for persist, could we clone that one and plumb it down here instead? (Compute intentionally works a little differently here because of some timely worker stuff)

Reply:
Oh, does storage have the same requirement as compute about all the timely workers observing changes in the same order w.r.t. all other messages on the controller protocol? If so, then yeah, this is the right thing to do, but I'd still love to fix the dep (e.g. by plumbing this down as a `Fn() -> ConfigSet` or something)

Reply:
Yeah they do!

Reply:
Don't think we actually fixed this dep yet? But I'm happy to do a followup PR to fix this and the compute one, so feel free to leave it unaddressed